app/Services/ModuleService.php

```diff
@@ -70,4 +76,54 @@ public function getAdminLinks(): array
     {
         return self::$adminLinks;
     }
+
+    public function createModule($request)
```

Comment:
I haven't been passing `$request` into these, only an array with the parameters needed. This makes it easier to test . But I can clean this up later.